### PR TITLE
adding multicore

### PR DESCRIPTION
**Pull request by wckdouglas**
@nolanmaier tested the integration of `BiocParallel` via rpy2 in #20 

This PR implements the solution into the `py_DESeq` object

### CODE REVIEW

File: diffexpr/py_deseq.py [hunk 1]
```diff
@@ -65,7 +68,8 @@ class py_DESeq2:
 
     """
 
-    def __init__(self, count_matrix, design_matrix, design_formula, gene_column="id"):
+    def __init__(self, count_matrix, design_matrix, design_formula, gene_column="id", threads=4):
```
Comment (nolanmaier) [inline diffexpr/py_deseq.py]: My understanding is that setting `threads=4` here will cause the default function to run in parallel with 4 threads. Maybe a better solution is to have the default setting run serially with `threads=1`. Then the user can update the threads param if they are interested in parallelization.

Comment (nolanmaier) [reply diffexpr/py_deseq.py]: maybe also implement a simple check that the threads parameter passed by the user is an integer.
```
if not isinstance(threads, int): 
    raise ValueError("threads must be an integer")
```

Comment (wckdouglas) [reply diffexpr/py_deseq.py]: good suggestions!!

**Comment by nolanmaier** [comment]
Looks good to me!

**Comment by wckdouglas** [comment]
@nolanmaier thanks for working with me on this!

**Pull request merged**